hack/generator/pkg/functions/resource_conversion_function.go

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package functions
+
+import (
+	"fmt"
+	"go/token"
+
+	"github.com/dave/dst"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astbuilder"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+)
+
+// ResourceConversionFunction implements conversions to/from our hub type
+// Existing PropertyAssignment functions are used to implement stepwise conversion
+//
+// Direct conversion from the hub type:
+//
+// func (r <receiver>) ConvertFrom(hub conversion.Hub) error {
```

Comment:
minor: use the `From|To` notation you used below to clarify that this can also happen in either direction?

Reply:
Or maybe because you aren't using it super consistently below either (and doing so might make the examples awkward for example in the error messages, etc) instead you could just preface this with something like: "Examples show the `From` direction. The only difference between the two is that all instances of `From` are replaced with `To` in the `To` direction." or something?

Reply:
Yeah FWIW I agree. In theory I like `<From|To>` but in practice it makes things a bit harder to read. I think my main thing here is just be consistent, if you don't use them that's fine you can just have a preface or something that explains _roughly_ how the other direction might look. The comments are meant to help us understand the code rather than be an exact, faithful replica of the template that is being applied so I think either is ok.

Reply:
At the moment I'm leaning towards keeping them in - it's still easier to parse the comment with `<To|From>` than the actual code, and it cues readers into the complexity that's present.